RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc

```diff
@@ -276,36 +276,7 @@ MuonShowerInformationFiller::hitsFromSegments(const GeomDet* geomDet,
 //
 // Find cluster
 //
-MuonTransientTrackingRecHit::MuonRecHitContainer 
-MuonShowerInformationFiller::findPhiCluster(MuonTransientTrackingRecHit::MuonRecHitContainer& muonRecHits, 
-                              const GlobalPoint& refpoint) const {
-
-  if ( muonRecHits.empty() ) return muonRecHits;
-
-  //clustering step by phi
-  float step = 0.05;
-  MuonTransientTrackingRecHit::MuonRecHitContainer result;
-
-  stable_sort(muonRecHits.begin(), muonRecHits.end(), AbsLessDPhi(refpoint));
 
-  for (MuonTransientTrackingRecHit::MuonRecHitContainer::const_iterator ihit = muonRecHits.begin(); ihit != muonRecHits.end() - 1; ++ihit) {
-      if (fabs(deltaPhi((*(ihit+1))->globalPosition().phi(), (*ihit)->globalPosition().phi() )) < step) {

```

Comment:
looking at the old code: I think most of the CPU is consumed by the transformation of the rechits from native local coordinates to the global coordinates.
If the main container is changed to {GlobalPoint, theta, phi, r}, and the precomputed phi and theta are used directly, the code should already become several orders of magnitude faster.